src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.exceptions.RequiredInformationMissingException;
+import duke.exceptions.UnknownCommandException;
+
+/**
+ * Class that has methods for handling user input
+ */
+public class Parser {
+
+    /**
+     * Parses the input typed in by user, returning a nicely formatted arraylist containing the command name as first
+     * index, and and relevant arguments from the input as the result.
+     *
+     * @param input
+     * @return List containing the parsedInput strings.
+     * @throws RequiredInformationMissingException if required info for a command not present in input.
+     * @throws UnknownCommandException             if command is unknown.
+     */
+    public static ArrayList<String> parseInput(String input) throws RequiredInformationMissingException,
+            UnknownCommandException {
+        ArrayList<String> parsedInput = new ArrayList<>();
+        String command = getCommandFromInput(input);
+        parsedInput.add(command);
+        if (command.equals("list")) {
+            parsedInput.add("list");
+        } else if (command.equals("mark")) {
+            if (input.length() < 6) {
+                throw new RequiredInformationMissingException("missing mark index");
+            }
```

Comment:
Since this method is a bit too lengthy, I suggest you move all these input checkings into separate small private methods that returns a boolean.

The rationale is that you can name the methods in a way that allows others to understand what it is checking. For example, I do not understand why is the need to check for `input.length() < 6`. Maybe putting it in a method named 'hasInputIndex()` will make more sense. 

Another suggestion is that, perhaps checking length isn't the most suitable way to check for missing index? Maybe you can try string matching to match for numerics only.